data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/ShardAcknowledgementManagerTest.java

```diff
@@ -83,4 +95,24 @@ void testIsExportDone() {
     void testShutdown() {
         assertDoesNotThrow(() -> shardAcknowledgementManager.shutdown());
     }
+
+    @Test
+    void testUpdateOwnershipForAllShardPartitions() throws Exception {
+        when(dynamoDBSourceConfig.getShardAcknowledgmentTimeout()).thenReturn(Duration.ofMinutes(15));
+        when(acknowledgementSetManager.create(any(Consumer.class), any(Duration.class))).thenReturn(acknowledgementSet);
+        
+        // Create acknowledgment set to add partition to checkpoints
+        shardAcknowledgementManager.createAcknowledgmentSet(streamPartition, "seq123", false);
+        
+        // Set lastCheckpointTime to past to trigger checkpoint interval
+        Field lastCheckpointTimeField = ShardAcknowledgementManager.class.getDeclaredField("lastCheckpointTime");
+        lastCheckpointTimeField.setAccessible(true);
+        lastCheckpointTimeField.set(shardAcknowledgementManager, Instant.now().minus(Duration.ofMinutes(5)));
```

Comment:
We have a class named `ReflectivelySetField` in `common-test` that you can use to do this.